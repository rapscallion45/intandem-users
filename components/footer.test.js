import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Footer from './footer';

describe('Footer', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    /* Act */
    const wrapper = shallow(<Footer />);

    /* Assert */
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Copyright date', () => {
    it('Should render this year', () => {
      /* Arrange */
      const today = new Date();
      const date = today.getFullYear();

      /* Act */
      const wrapper = render(<Footer />);

      /* Assert */
      expect(wrapper.getByText(date)).toBeInTheDocument();
    });
  });

  describe('Linkedin link', () => {
    it('Should render link to Linkedin profile', () => {
      /* Arrange */
      const linkText = 'Carl Scrivener';
      const linkUrl = 'https://www.linkedin.com/in/carlscrivener/';

      /* Act */
      const wrapper = render(<Footer />);

      /* Assert */
      expect(wrapper.getByText(linkText).closest('a')).toHaveAttribute('href', linkUrl);
    });
  });
});
