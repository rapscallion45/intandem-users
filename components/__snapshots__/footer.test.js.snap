// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Renders correctly enzyme 1`] = `
<ForwardRef(Box)
  component="footer"
  sx={
    Object {
      "backgroundColor": [Function],
      "mt": "auto",
      "px": 2,
      "py": 3,
    }
  }
>
  <ForwardRef(Container)
    maxWidth="sm"
  >
    <ForwardRef(Typography)
      align="center"
      variant="body1"
    >
      This is a test application intended for demonstration purposes only.
    </ForwardRef(Typography)>
    <Copyright />
  </ForwardRef(Container)>
</ForwardRef(Box)>
`;
